VFS Archive support

ajmeese7 opened the pull request:
Meant to close out #64, pending the merge of https://github.com/os-js/osjs-gui/pull/40 and https://github.com/os-js/osjs-filemanager-application/pull/41. Will result in the resolution of https://github.com/os-js/osjs-filemanager-application/issues/39 as well.

Opening as a draft because there are still a few slight modifications that I want to make, most notably a `try/catch` for the `compress` action that deletes the ZIP file on an error rather than leaving behind a corrupt artifact. Also there is one test (noted with a TODO) that I implemented that still doesn't work and that I have been unable to figure out, so some help with that would be appreciated.

ajmeese7 committed:
Added .editorconfig

ajmeese7 committed:
Merge branch 'os-js:master' into feature/archive-support

ajmeese7 committed:
Initial implementation

ajmeese7 commented:
Also yes, as the CI pointed out, I didn't update the lockfile yet either.

ajmeese7 committed:
Updated package lock

ajmeese7 committed:
Delete archive on compress error

ajmeese7 committed:
Resolved eslint error

ajmeese7 commented:
I don't know why the CI linter doesn't like my switch statement but I don't see anything wrong with it, the indentation looks correct to me. @andersevenrud could you weigh in when you get a chance?

I could change the `eslint` rules to the following if you want, this should solve it: `indent: ['error', 2, { SwitchCase: 1 }],`

andersevenrud commented:
> I could change the eslint rules to the following if you want, this should solve it

Feel free to do so. I don't think I have a switch statement in the codebase anywhere else, so this has never popped up for me :)

ajmeese7 committed:
chore: linting

ajmeese7 committed:
feat: deletes original file after adding to archive
Also removed deletion of archive after extraction, inline with how operating systems like Windows handle archive extraction.

ajmeese7 committed:
feat: support sanitization of file path arrays

ajmeese7 committed:
fix: improved tests
